Added PrivateLink service documentation: (#7425)
magento/devdocs: src/cloud/project/privatelink-service.md

---
group: cloud-guide
title: PrivateLink service
functional_areas:
  - Cloud
  - Setup
---

Magento supports PrivateLink connections for Cloud customers to establish secure, private communication between {{site.data.var.ece}} environments and external systems. Both the Magento application and external systems must be accessible through private VPC endpoints hosted within the same Cloud region (AWS or Azure).

## Features and support

The AWS PrivateLink service for Magento Cloud customers includes the following features and support:

-  A secure connection between a customer Virtual Private Cloud (VPC) and the Magento VPC within the same Cloud region.
-  Support for unidirectional or bidirectional communication between Magento and Customer VPCs.
-  Service enablement–
   -  Open communication on the necessary ports in the Magento Cloud environment
   -  Establish the initial connection between the customer and Magento VPC
   -  Troubleshoot connection issues during enablement

## Limitations

-  Support for PrivateLink is available on Pro plan Production and Staging environments only. It is not available on local or integration environments, or on Starter plan projects.

> -  You cannot establish SSH connections using PrivateLink. Due to infrastructure requirements, Magento cannot open Port 22 for communication between VPCs.

Reviewer: This is in our style guide, which was why I mentioned it. Ultimately it is writer discretion.

## Request PrivateLink enablement

{:.bs-callout-warning}
Enabling PrivateLink can take up to 5 business days. Providing incomplete, or inaccurate information when you request enablement can delay the process.

### Prerequisites

-  A Cloud account (AWS or Azure) in the same region as the {{site.data.var.ece}} instance
-  A VPC in the customer environment that hosts the services to connect via PrivateLink
-  For bidirectional PrivateLink connections, your must configure the PrivateLink service in your VPC environment before requesting Magento PrivateLink enablement. See [Customer set up for bidirectional PrivateLink communication][#customer-set-up-for-bidirectional-privateLink-communication].
-  Gather the following data required for PrivateLink enablement:

   -  **Cloud account number** (AWS or Azure)–Must be in the same region as the Magento Cloud instance
   -  **Cloud region**–Provide the Cloud region where the account is hosted for verification purposes
   -  **Services and communication ports**–Magento must open ports to enable service communication between VPCs, for example _Webserver, HTTP port 80_, _SFTP port 2222_

>    -  **Magento Cloud Project ID**

Reviewer: The only one without a statement. Maybe can identify where to get the ID just so there is something there.

>    -  **Connection Type**–Specify unidirectional or bidirectional for connection type

Reviewer: ```suggestion
   -  **Connection Type**–Specify unidirectional or  bidirectional for connection type
```

>    -  **Service Name**–Service name for the VPC service endpoint that Magento must connect to. Required only to enable bidirectional PrivateLink connections.

Reviewer: ```suggestion
   -  **Service Name**–Service name for the VPC service endpoint. Magento requires this only to enable bidirectional PrivateLink connections.
```

1. Additional steps to enable bidirectional connections:

   -  **Magento infrastructure team** uses the endpoint service name supplied by the customer to submit a connection request from the Magento VPC to the customer Network Load Balancer (NLB).

   -  **Customer** approves the request to complete the setup. See the Cloud platform documentation for instructions:

      -  For AWS, see [Accepting and rejecting interface endpoint connection requests][].
      -  For Azure, see [Manage connection requests][].

   -  **Customer** verifies the connection to and from the Magento VPC.

## Customer set up for bidirectional PrivateLink communication

The customer VPC must have a the following resources available to support bidirectional PrivateLink connections:

-  A Network Load Balancer
-  An endpoint service configuration that enables access to a service from your VPC environment
-  An [interface endpoint][] (AWS) or [private endpoint][] (Azure) that allows Magento to connect to endpoint services hosted in your VPC

If these resources are not available in the customer VPC, you must sign into your Cloud platform account to add the configuration.

> -  Amazon VPC console– ```https://console.aws.amazon.com/vpc/```

Reviewer: Not sure why code block ticks are used here. Best to use single ticks unless they are posing a strange problem.

-  Azure portal– `https://portal.azure.com`

See your Cloud platform documentation for PrivateLink set up instructions:

> -  **AWS PrivateLink  documentation**

Reviewer: Noticed extra space

```suggestion
-  **AWS PrivateLink documentation**
```
